### ISSUE TITLE
addblock not functioning with latest update

### ISSUE THREAD

**Issue by MoeFwacky** [amazingr4b/TBN-Plex#1]
Since a fresh setup of the most recent version, addblock no longer works. I have a raspberry pi running Rasplex, another which hosts this script and the Plex server on my Windows 10 machine.

Output below.

Traceback (most recent call last):
  File "/home/pi/hasystem/system.py", line 3464, in <module>
    say = addblock(name, title)
  File "/home/pi/hasystem/system.py", line 365, in addblock
    cur.execute(command2)
sqlite3.OperationalError: no such table: TVshowlist

**Comment by amazingr4b** [comment]
I should have probably put a note on there saying people would need to run
the setup script and update their TV library to generate that table. I
changed quite a bit, and many things will run checks against it. Get the
new ones if you didnt and Run those 2 scripts and then try again. Let me
know if it still gives that error.

**Comment by MoeFwacky** [comment]
I tried that and I also tried scrubbing the folder entirely and running setup from scratch. In both cases I still get that error.

**Comment by amazingr4b** [comment]
Thanks for the info. I'll see if I can figure out what's going on.

**Comment by MoeFwacky** [comment]
This may be related. I noticed when trying to run upddatedb_pi.py that it appears a / is missing in the output urls

http://192.168.1.120:32400/library/metadata1444/allLeaves
http://192.168.1.120:32400/library/metadata1136/allLeaves
http://192.168.1.120:32400/library/metadata1665/allLeaves
http://192.168.1.120:32400/library/metadata2361/allLeaves

If I'm not mistaken, previously there used to be a / between metadata and the numbers.

**Comment by MoeFwacky** [comment]
Quick update. I deleted the .myplex.db file and ran upddatedb_pi.py. A new database was generated and all the shows properly added to it, but attempting to create a new block still results in the no such table error.

**Comment by amazingr4b** [comment]
I know what the problem is. The version of the upddatedb_pi.py file is not
the correct one, and is missing the actions to create and populate the
TVshowlist table. I'm double checking the version I have running it on a
fresh instance now to make sure it works as intended. Once that is done
running I'll post it.

**Comment by amazingr4b** [comment]
I posted some new files to fix that missing TVshowlist table error. ...  As
far as the / in the link, I'm guessing when you copied in one of the links
you cut off the last character, which is a /. I added a reset option to the
upddatedb_pi.py script so if that happens, of even needs changing it can
easily be done.

Thanks for testing,

**Comment by MoeFwacky** [comment]
A couple of new issues have cropped up. I've double checked that my entries are correct, and it looks like they are currently.

I tried a fresh install, but receive "Error adding" on what looks like every episode of every show.

Also, about halfway through (in the M's alphabetically), it crashes with the following:

Traceback (most recent call last):
  File "/home/pi/hasystem/upddatedb_pi.py", line 984, in <module>
    gettvshows()
  File "/home/pi/hasystem/upddatedb_pi.py", line 722, in gettvshows
    cur.execute('SELECT * FROM shows WHERE TShow LIKE \'' + TShow + '\' AND Tnum LIKE \'' + str(Tnum) + '\'')
sqlite3.OperationalError: near "s": syntax error
If you needed that entry for cron it was: @reboot python /home/pi/hasystem/piplaystate.py &

**Comment by amazingr4b** [comment]
Sounds like your shows table was built using the bad version that had the
extra column. I fixed that in the file I posted early this morning. Unless
you have an editor and can drop "Duration" from the shows table you'll
likely need to start with a fresh dB. The other problem is probably caused
by a character in show title that sql does not like. Some characters cause
it to error, and I've not built exceptions for but the few I've run across.
On Jul 9, 2016 6:49 PM, "MoeFwacky" <notifications@github.com> wrote:

A couple of new issues have cropped up. I've double checked that my entries
are correct, and it looks like they are currently.

I tried a fresh install, but receive "Error adding" on what looks like
every episode of every show.

Also, about halfway through (in the M's alphabetically), it crashes with
the following:

Traceback (most recent call last):
File "/home/pi/hasystem/upddatedb_pi.py", line 984, in
gettvshows()
File "/home/pi/hasystem/upddatedb_pi.py", line 722, in gettvshows
cur.execute('SELECT * FROM shows WHERE TShow LIKE \'' + TShow + '\' AND
Tnum LIKE \'' + str(Tnum) + '\'')
sqlite3.OperationalError: near "s": syntax error
If you needed that entry for cron it was: @reboot
<https://github.com/reboot> python /home/pi/hasystem/piplaystate.py &

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/amazingr4b/TBN-Plex/issues/1#issuecomment-231562135>,
or mute the thread
<https://github.com/notifications/unsubscribe/ASgMDlvANVaYagLvRpSHu5yWrJDJl7A2ks5qUDOKgaJpZM4JIo6_>
.
